internal/overlord/overlord.go

```diff
@@ -140,13 +141,23 @@ func (o *Overlord) addManager(mgr StateManager) {
 }
 
 func loadState(statePath string, restartBehavior RestartBehavior, backend state.Backend) (*state.State, error) {
-
 	timings := timing.Start("", "", map[string]string{"startup": "load-state"})
 
 	curBootID, err := osutil.BootID()
 	if err != nil {
 		return nil, fmt.Errorf("fatal: cannot find current boot ID: %v", err)
 	}
+	// If pebble is PID 1 we don't care about /proc/sys/kernel/random/boot_id
+	// as we are most likely running in a container. LXD mounts it's own boot_id
+	// to correctly emulate the boot_id behaviour of non-containerized systems.
+	// Within containerd/docker, boot_id is consistent with the host, which provides
+	// us no context of restarts.
+	if os.Getpid() == 1 {
+		curBootID, err = strutil.UUID()
+		if err != nil {
+			return nil, fmt.Errorf("fatal: failed to generate psuedo boot-id: %v", err)
```

Comment:
As a design convention I've tried to keep, for consistency in wording all variants such as "failed to", "unable to", "could not", "can not", "error while", etc, are all always spelled consistently as _"cannot"_. But this is nitpick-level detail, and if this is the only thing in this PR I'll just fix it later.

Reply:
I like this, thanks for the suggestion.